### PR TITLE
[frontend] Add ability to override confidence per entity type in Group (#6878)

### PR DESCRIPTION
**Pull request by JeremyCloarec**
### Proposed changes

*  [x]  add Capabilities tab to group edition
*  [ ] fix behavior in user and group capabilities tab:
   *  [x]  prevent adding multiple overrides without an entity type
   *  [x] add current max_confidence of user/group to initial override confidence value
   *  [x]  prevent adding the same entity type twice
   *  [x]  restrict entity type selection 
   *  [ ]  refactor user and group component to prevent code duplication

### Related issues

* #6878 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/users/edition/UserEditionConfidence.tsx [hunk 1]
```diff
@@ -187,10 +187,10 @@ const UserEditionConfidence: FunctionComponent<UserEditionConfidenceProps> = ({
                   <IconButton
                     color="primary"
                     aria-label="Add"
-                    onClick={() => arrayHelpers.push({ entity_type: '', max_confidence: 0 })}
+                    onClick={() => arrayHelpers.push({ entity_type: '', max_confidence: user.effective_confidence_level && user.effective_confidence_level.max_confidence ? user.effective_confidence_level.max_confidence : 0 })}
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/users/edition/UserEditionConfidence.tsx]: Badass cette ligne de code

Could be a function or at least separate into multiple lines?

File: opencti-platform/opencti-front/src/private/components/settings/users/edition/ConfidenceOverrideField.tsx [hunk 2]
```diff
@@ -26,18 +26,21 @@ interface UserConfidenceOverridesFieldComponentProps
...
   field,
   index,
   onDelete,
   onSubmit,
+  currentOverrides,
 }) => {
   const { t_i18n } = useFormatter();
   const theme = useTheme<Theme>();
   const { availableEntityTypes } = useSchema();
+  const entityTypesToOverride = availableEntityTypes.filter((entity_type) => entity_type.type !== 'entity_Stix-Meta-Objects' && entity_type.type !== 'entity_Stix-Cyber-Observables');
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/users/edition/ConfidenceOverrideField.tsx]: Badass cette ligne de code

Could be separate into multiple lines?

**Comment by lndrtrbn** [comment]
image

When opening the dropdown to select an entity type to override

**Comment by lndrtrbn** [comment]
image

Should it be "Confiance" for French translation?


@SarahBocognano : ===> Same for the "Probably True"

**Review by SarahBocognano — APPROVED** [review]
Tested ✅

**Review by lndrtrbn — CHANGES_REQUESTED** [review]
cf above

**Comment by lndrtrbn** [comment]
App crash when cleaning the entity type field

https://github.com/OpenCTI-Platform/opencti/assets/33682259/529fa3c2-53d5-4a96-975f-d8a18fc88770

EDIT: also crashes on user edit

**Comment by lndrtrbn** [comment]
image

Feels odd to have "00" instead of "0"

**Comment by Archidoit** [comment]
tested OK 👍

**Review by Archidoit — APPROVED** [review]

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**